code/game/objects/explosion.dm

```diff
@@ -146,9 +146,8 @@ var/explosion_shake_message_cooldown = 0
 	var/list/cached_exp_block = CalculateExplosionBlock(affected_turfs)
 
 	for(var/turf/T in affected_turfs)
-		if(whitelist)
-			if(T in whitelist)
-				continue
+		if(whitelist && (T in whitelist))
```

Comment:
do whitelist.len in both cases, it's more reliable because it counts the number of entries instead of whether the list exists.

Reply:
There's no need to check `len` since there's nothing wrong with searching an empty list